example-crates/example_rpc/src/main.rs

```diff
@@ -0,0 +1,308 @@
+            (_, Some(_), None) => panic!("rpc auth: missing rpc_pass"),
+            (_, None, Some(_)) => panic!("rpc auth: missing rpc_user"),
+        }
+    }
+}
+
+#[derive(Subcommand, Debug, Clone)]
+enum RpcCommands {
+    /// Scans blocks via RPC (starting from last point of agreement) and stores/indexes relevant
+    /// transactions
+    Scan {
+        /// Starting block height to fallback to if no point of agreement if found
+        #[clap(env = "FALLBACK_HEIGHT", long, default_value = "0")]
+        fallback_height: u32,
+        /// The unused-scripts lookahead will be kept at this size
+        #[clap(long, default_value = "10")]
+        lookahead: u32,
+        /// Whether to be live!
+        #[clap(long, default_value = "false")]
+        live: bool,
+        #[clap(flatten)]
+        rpc_args: RpcArgs,
+    },
+    /// Create and broadcast a transaction.
+    Tx {
+        value: u64,
+        address: Address<address::NetworkUnchecked>,
+        #[clap(short, default_value = "bnb")]
+        coin_select: CoinSelectionAlgo,
+        #[clap(flatten)]
+        rpc_args: RpcArgs,
+    },
+}
+
+impl RpcCommands {
+    fn rpc_args(&self) -> &RpcArgs {
+        match self {
+            RpcCommands::Scan { rpc_args, .. } => rpc_args,
+            RpcCommands::Tx { rpc_args, .. } => rpc_args,
+        }
+    }
+}
+
+fn main() -> anyhow::Result<()> {
+    let sigterm_flag = start_ctrlc_handler();
+
+    let (args, keymap, index, db, init_changeset) =
+        example_cli::init::<RpcCommands, ChangeSet>(DB_MAGIC, DB_PATH)?;
+
+    let graph = Mutex::new({
+        let mut graph = IndexedTxGraph::new(index);
+        graph.apply_additions(init_changeset.indexed_additions);
+        graph
+    });
+
+    let chain = Mutex::new(LocalChain::from_changeset(init_changeset.chain_changeset));
+
+    let rpc_cmd = match args.command {
+        example_cli::Commands::ChainSpecific(rpc_cmd) => rpc_cmd,
+        general_cmd => {
+            let res = example_cli::handle_commands(
+                &graph,
+                &db,
+                &chain,
+                &keymap,
+                args.network,
+                |_| Err(anyhow::anyhow!("use `tx` instead")),
+                general_cmd,
+            );
+            db.lock().unwrap().commit()?;
+            return res;
+        }
+    };
+
+    let rpc_client = {
+        let a = rpc_cmd.rpc_args();
+        Client::new(
+            &a.url,
+            match (&a.rpc_cookie, &a.rpc_user, &a.rpc_password) {
+                (None, None, None) => Auth::None,
+                (Some(path), _, _) => Auth::CookieFile(path.clone()),
+                (_, Some(user), Some(pass)) => Auth::UserPass(user.clone(), pass.clone()),
+                (_, Some(_), None) => panic!("rpc auth: missing rpc_pass"),
+                (_, None, Some(_)) => panic!("rpc auth: missing rpc_user"),
+            },
+        )?
+    };
+
+    match rpc_cmd {
+        RpcCommands::Scan {
```

Comment:
Maybe just `sync` instead of `scan` since we use that for scanning addresses. This is rather syncing local state with remote state like the sync commands.